sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -477,11 +483,33 @@ selectClause
     ;
 
 setClause
-    : SET assign (',' assign)*
+    : SET assignmentList
     ;
 
-assign
-    : key=multipartIdentifier EQ value=expression
+matchedClause
+    : WHEN MATCHED (AND matchedCond=booleanExpression)? THEN matchedAction
+    ;
+notMatchedClause
+    : WHEN NOT MATCHED (AND notMatchedCond=booleanExpression)? THEN notMatchedAction
+    ;
+
+matchedAction
+    : DELETE
+    | UPDATE SET ASTERISK
+    | UPDATE SET assignmentList
+    ;
+
+notMatchedAction
+    : INSERT ASTERISK
+    | INSERT '(' columns=qualifiedNameList ')' VALUES values=valueList
+    ;
+
+assignmentList
+    : assignment (',' assignment)*
+    ;
+
+assignment
+    : key=qualifiedName EQ value=expression
```

Comment:
previous it's `key=multipartIdentifier EQ value=expression`, why change to `qualifiedName`?

Reply:
`multipartIdentifier` and `qualifiedName` are essentially identical except that `multipartIdentifier` handles some error msg processing. I found most of columns/functions are using `qualifiedName` while tables are using `multipartIdentifier`, so I changed it to `qualifiedName`.